packages/crayons-core/src/components/icon/readme.md

```diff
@@ -23,28 +30,184 @@ import ReactDOM from "react-dom";
 import { FwIcon } from "@freshworks/crayons/react";
 function App() {
   return (<div>
-  <FwIcon name="add-contact" size="18" color="green"></FwIcon>
+  <FwIcon name="add-contact" size="18" color="green" ></FwIcon>
  </div>);
 }
 ```
 </code-block>
+
+<code-block title="Using Intersection Observer">
+```html 
+HTML
+<fw-icon name="add-contact" size="18" color="green" x-root-margin = "80px" lazy></fw-icon>
+<fw-icon name="add-contact" size="18" color="green" lazy></fw-icon>
+React
+<FwIcon name="add-contact" size="18" color="green" xRootMargin = "80px" lazy></FwIcon>
+```
+</code-block>
 </code-group>
 
-### Icons
+### Intersection Observer
+
+Use prop 'lazy' to enable Intersection-Observer. By deafult it is disabled. You may choose to give the intersection root-margin for icons i.e via prop 'x-root-margin' as preloading threshold in a viewport.Default value is '50px'.
+fw-icon also implements fetch API Memoization so any icon once fetched will not be requested again and will be available throughout the session/pages.
+
+## Crayons Icon Assets 
 
-The following are the icons supported:
+The following icons are presently part of the Crayons-Icon library. These are optimized using SVGO.
+Use the name of an icon as listed below it. For JS Imports, you may click to copy the imports.
 
 <IconGallery/>
+
+# Icon Library (@freshworks/crayons-icon)
+
+Crayons Icon is now available as '@freshworks/crayons-icon' Library. This encapsulates all Icon Tooling icon exports. Following is implemented via the Lib.
+
+1. JS Exports of SVG Icon to enable Tree-Shaking for inline-svg. This is a useful feature where Dev Team can choose to do something offline with SVGs.
+2. Enable Crayons-Icon lib to support external icon libraries. The development team can register external icon libraries via register API. See usage docs.
+3. Optimized SVGs using SVGO via .yml configuration. The development team can extend the .yml file and use it in their app for optimizing their custom icons. 
```

Comment:
Can you provide an example for this ?, How can a consuming app extend and optimize their custom svgs?.

Reply:
Will need to address after PR Merge.

Reply:
Added CLI Showcase